src/locktower.rs

```diff
@@ -0,0 +1,172 @@
+use std::collections::VecDeque;
+pub struct VoteLock {
+    /// Fork id, same as slot id. There can only be 1 fork per slot, since the empty PoH ticks
+    /// cannot be voted on.  A validator cannot vote on a fork unless that fork is connected to
+    /// a trunk. The starting trunk is the initial `finalized_fork`.
+    fork: u64,
+    /// Lockout is in terms of slots, so the slot height or fork id to unroll this
+    /// vote must be greater than fork + lockout.
+    lockout: u64,
```

Comment:
Do we really need one of these per `VoteLock`?  Seems like there just needs to be a `u8` in LockTower and then that in combination with its position in the tower is sufficient to calculate `lockout`.  So:

```
lockout == 1 << locktower.lockout << lock_index
```

meaning that ever time you want to double the lockout of all locks in a tower, you just `locktower.lockout += 1`

Reply:
@garious what about cleared votes?

Reply:
lockouts don't all double at the same time, only those whose lockouts are equal to the next vote, as described here:
https://github.com/solana-labs/solana/blob/master/book/src/fork-selection.md#rollback